rivertest/worker.go

```diff
@@ -0,0 +1,186 @@
+	t.Helper()
+
+	config = config.WithDefaults()
+	client, err := river.NewClient(driver, config)
+	require.NoError(t, err)
+
+	return &Worker[T, TTx]{
+		client: client,
+		config: config,
+		worker: worker,
+	}
+}
+
+func (w *Worker[T, TTx]) Work(ctx context.Context, t *testing.T, args T) error {
+	t.Helper()
+	return w.WorkOpts(ctx, t, args, nil)
+}
+
+func (w *Worker[T, TTx]) WorkJob(ctx context.Context, t *testing.T, job *river.Job[T]) error {
+	t.Helper()
+
+	// populate river client into context:
+	ctx = WorkContext(ctx, w.client)
+
+	doInner := execution.MiddlewareChain(
+		w.config.WorkerMiddleware,
+		w.worker.Middleware(job),
+		func(ctx context.Context) error { return w.worker.Work(ctx, job) },
+		job.JobRow,
+	)
+
+	jobTimeout := valutil.FirstNonZero(w.worker.Timeout(job), w.config.JobTimeout)
+	ctx, cancel := execution.MaybeApplyTimeout(ctx, jobTimeout)
+	defer cancel()
+
+	return doInner(ctx)
+}
+
+func (w *Worker[T, TTx]) WorkOpts(ctx context.Context, t *testing.T, args T, opts *river.InsertOpts) error {
+	t.Helper()
+
+	job := makeJobFromArgs(t, w.config, args, opts)
+	return w.WorkJob(ctx, t, job)
+}
+
+var idSeq int64 = 0 //nolint:gochecknoglobals
+
+func nextID() int64 {
+	return atomic.AddInt64(&idSeq, 1)
+}
+
+func makeJobFromArgs[T river.JobArgs](t *testing.T, config *river.Config, args T, opts *river.InsertOpts) *river.Job[T] {
+	t.Helper()
+
+	if opts == nil {
+		opts = &river.InsertOpts{}
+	}
+
+	encodedArgs, err := json.Marshal(args)
+	require.NoError(t, err)
+
+	// Round trip the args to validate JSON marshalling/unmarshalling on the type.
+	var decodedArgs T
+	if err = json.Unmarshal(encodedArgs, &decodedArgs); err != nil {
+		t.Fatalf("failed to unmarshal args: %s", err)
+	}
+
+	// Extract any job insert opts from args type if present
+	var jobInsertOpts river.InsertOpts
+	if argsWithOpts, ok := any(args).(river.JobArgsWithInsertOpts); ok {
+		jobInsertOpts = argsWithOpts.InsertOpts()
+	}
+
+	now := valutil.FirstNonZero(opts.ScheduledAt, jobInsertOpts.ScheduledAt, time.Now())
+
+	internalUniqueOpts := (*dbunique.UniqueOpts)(&opts.UniqueOpts)
+	insertParams := &rivertype.JobInsertParams{
+		Args:        args,
+		CreatedAt:   &now,
+		EncodedArgs: encodedArgs,
+		Kind:        args.Kind(),
+		MaxAttempts: valutil.FirstNonZero(opts.MaxAttempts, jobInsertOpts.MaxAttempts, config.MaxAttempts, river.MaxAttemptsDefault),
+		Metadata:    sliceutil.FirstNonEmpty(opts.Metadata, jobInsertOpts.Metadata, []byte(`{}`)),
+		Priority:    valutil.FirstNonZero(opts.Priority, jobInsertOpts.Priority, river.PriorityDefault),
+		Queue:       valutil.FirstNonZero(opts.Queue, jobInsertOpts.Queue, river.QueueDefault),
+		State:       rivertype.JobStateAvailable,
+		Tags:        sliceutil.FirstNonEmpty(opts.Tags, jobInsertOpts.Tags, []string{}),
+	}
+	// TODO: use exposed client config Time when added
```

Comment:
Timely and relevant to #751: here's a place within River where I'm wanting to set the time generator explicitly for testing purposes (though I need to reach into the internal dbunique package as well).